src/service-implementations/lf-repo-tree-node-service.ts

```diff
@@ -272,55 +281,104 @@ export class LfRepoTreeNodeService implements LfTreeNodeService {
+              throw new Error('Unexpected shortcut targetType');
+          }
+          break
+      default:
+        throw new Error('Unsupported entry type');
+    }
+    this.setColumnAttributesForEntry(entry, treeNode);
+    this.setTargetIdAndTypeForShortcut(entry, treeNode);
+    return treeNode;
+  }
+
+  private getIconForEntry(entryType: EntryType, parent?: LfRepoTreeNode, extension?: string): string {
+    switch (entryType) {
+      case EntryType.Folder: {
+        if (parent?.entryType === EntryType.RecordSeries ||
+          (parent?.entryType === EntryType.Shortcut && parent?.targetType === EntryType.RecordSeries)) {
+          return IconUtils.getDocumentIconUrlFromIconId('recordfolder-20');
+        }
+        else {
+          return IconUtils.getDocumentIconUrlFromIconId('folder-20');
+        }
       }
-      else if (shortcut.targetType === EntryType.RecordSeries) {
-        treeNode = this.createFolderNode(shortcut, repoName, parent);
+      case EntryType.RecordSeries: {
+        return IconUtils.getDocumentIconUrlFromIconId('recordseries-20');
       }
-      else {
-        throw new Error('Unexpected shortcut targetType');
+      case EntryType.Document: {
+        if (extension) {
+          console.log("extension: " + extension);
+          const iconId = IconUtils.getDocumentIconIdFromExtension(extension);
+          return IconUtils.getDocumentIconUrlFromIconId(iconId);
+        }
+        else {
+          return IconUtils.getDocumentIconUrlFromIconId('document-20');
+        }
+      }
+      case EntryType.Shortcut: {
+        return IconUtils.getDocumentIconUrlFromIconId('shortcut-overlay');
+      }
+      default: {
+        throw new Error('unsupported entry type');
       }
-    } else if (entry.entryType === EntryType.RecordSeries) {
-      treeNode = this.createFolderNode(entry, repoName, parent);
-    }
-    else {
-      throw new Error('Unsupported entry type');
     }
-    this.setNodeProperties(treeNode, entry, parent);
-    return treeNode;
   }
 
   private isViewable(entry: Entry): boolean {
     return (this.viewableEntryTypes?.includes(entry.entryType) && entry.entryType !== EntryType.Shortcut)
       || (this.viewableEntryTypes?.includes(EntryType.Shortcut) && entry.entryType === EntryType.Shortcut && this.viewableEntryTypes?.includes((entry as Shortcut).targetType));
   }
 
-  private createLeafNode(entry: Entry, parent?: LfRepoTreeNode): LfRepoTreeNode {
-    const path = parent ? PathUtils.combinePaths(parent.path, entry.name) : entry.fullPath;
-    if (!path) {
-      throw new Error('entry fullPath is undefined');
+  private createLeafNode(entry: Entry, parent: LfRepoTreeNode): LfRepoTreeNode {
+    let entryName: string | undefined = entry.name;
+    if (!entryName || entryName.length === 0) {
+      entryName = entry.id.toString();
+    }
+    const path = this.getFullPath(parent, entry.name);
+    let icon : string | string[];
+    const entryType = entry.entryType;
```

Comment:
no need to declare then assign icon

Reply:
this is for shortcut support. if the entry is a shortcut, icon is an array.

Reply:
but you always assign it first to this.getSingleIconForEntryType(entryType, parent, extension)
so you can do
let icon: string | string[] = this.getSingleIconForEntryType(entryType, parent, extension);

Reply:
or maybe that should be a different variable that is used to "calculate"  for shortcut rather than assigning to the same one